### DISCUSSION TITLE
Shall we add keyboard bindings and which ones

### DISCUSSION THREAD

**Discussion by estruyf** [estruyf/vscode-front-matter#288] [General]
The great thing about Visual Studio Code is that you can configure your own keyboard bindings, but should we add default ones? for instance:

- Open dashboard
- Insert image
- Insert snippet

**Questions**:

- Shall we add keyboard bindings?
- For what type of actions do you want to use keyboard bindings?
- Which keyboard bindings would you bind to which type of actions?

**Comment by michaeltlombardi** [comment]
I think keybindings for those three actions in particular would be hugely useful to me - otherwise I have to open the command palette to take those actions.

I think these could be viable options:

- <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>V</kbd> to insert a snippet and <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>I</kbd> to insert an image
- <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>V</kbd>+<kbd>S</kbd> to insert a snippet and <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>V</kbd>+<kbd>I</kbd> to insert an image
- <kbd>Alt</kbd>+<kbd>D</kbd> for the dashboard

But that's just on a cursory noodle, not extensive check. I think it _could_ be useful to have hotkeys for open/closing the Front Matter side bar or to reopen a data file in the dashboard view (but that's maybe more easily addressable as a right-click action on data files). I think most other actions I take are things I might make a keybinding for but aren't necessarily worth doing for others (check diagnostics, reload config).

**Reply by estruyf** [reply]
Makes sense, will have to bind these for my environment to see how it works, but it would make it easier. Although it can be configured manually, giving you a default might make adoption smoother.

**Reply by estruyf** [reply]
Created an issue for this question: https://github.com/estruyf/vscode-front-matter/issues/502
